config/v1/types_authentication.go

```diff
@@ -607,7 +607,15 @@ type OIDCClientReference struct {
 // +kubebuilder:validation:XValidation:rule="has(self.prefixPolicy) && self.prefixPolicy == 'Prefix' ? (has(self.prefix) && size(self.prefix.prefixString) > 0) : !has(self.prefix)",message="prefix must be set if prefixPolicy is 'Prefix', but must remain unset otherwise"
 // +union
 type UsernameClaimMapping struct {
-	TokenClaimMapping `json:",inline"`
+	// claim is a required field that configures the JWT token
+	// claim whose value is assigned to the cluster identity
+	// field associated with this mapping.
+	//
+	// claim must not be an empty string ("").
+	//
+	// +required
+	// +kubebuilder:validation:MinLength:=1
+	Claim string `json:"claim"`
```

Comment:
Is there a reasonable maximum length that we could add for this?

Reply:
I don't believe there are any restrictions on the claim name length for JWTs in general, but I do know the recommendation it to keep them pretty short in general.

Out of an abundance of caution, something like 256 is probably fine

Reply:
Added this in the latest push and added a ratcheting test for it